perfkitbenchmarker/providers/openstack/os_virtual_machine.py

```diff
@@ -391,6 +394,17 @@ def CreateScratchDisk(self, disk_spec):
 
     self._CreateScratchDiskFromDisks(disk_spec, disks)
 
+  def GetResourceMetadata(self):
+    """Returns a dict containing metadata about the VM.
+
+    Returns:
+      dict mapping string property key to value.
+    """
+    result = super(OpenStackVirtualMachine, self).GetResourceMetadata()
+    if FLAGS.openstack_post_provisioning_script:
```

Comment:
flag values should not be reused. Store the value as a object attribute, i.e. self.post_provisioning_script = FLAGS.openstack_post_provisioning_script during __init__ and then use self.post_provisioning_script in later calls.

Reply:
Thank you @jellyfishcake, I have updated patchset.